crates/script/src/sequence.rs

```diff
@@ -26,6 +23,86 @@ use std::{
 };
 use yansi::Paint;
 
+/// Returns the commit hash of the project if it exists
+pub fn get_commit_hash(root: &Path) -> Option<String> {
+    Git::new(root).commit_hash(true, "HEAD").ok()
+}
+
+pub enum ScriptSequenceKind {
+    Single(ScriptSequence),
+    Multi(MultiChainSequence),
+}
+
+impl ScriptSequenceKind {
+    pub fn save(&mut self, silent: bool, save_ts: bool) -> Result<()> {
+        match self {
+            ScriptSequenceKind::Single(sequence) => sequence.save(silent, save_ts),
+            ScriptSequenceKind::Multi(sequence) => sequence.save(silent, save_ts),
+        }
+    }
+
+    pub fn sequences(&self) -> impl Iterator<Item = &ScriptSequence> {
+        match self {
+            ScriptSequenceKind::Single(sequence) => std::slice::from_ref(sequence).iter(),
+            ScriptSequenceKind::Multi(sequence) => sequence.deployments.iter(),
+        }
+    }
+
+    pub fn sequeneces_mut(&mut self) -> impl Iterator<Item = &mut ScriptSequence> {
+        match self {
+            ScriptSequenceKind::Single(sequence) => std::slice::from_mut(sequence).iter_mut(),
+            ScriptSequenceKind::Multi(sequence) => sequence.deployments.iter_mut(),
+        }
+    }
+
+    pub fn sequences_len(&self) -> usize {
+        match self {
+            ScriptSequenceKind::Single(_) => 1,
+            ScriptSequenceKind::Multi(sequence) => sequence.deployments.len(),
+        }
+    }
+
+    pub fn get_sequence_mut(&mut self, index: usize) -> Option<&mut ScriptSequence> {
+        match self {
+            ScriptSequenceKind::Single(sequence) => {
+                if index == 0 {
+                    Some(sequence)
+                } else {
+                    None
+                }
+            }
+            ScriptSequenceKind::Multi(sequence) => sequence.deployments.get_mut(index),
+        }
+    }
+
+    /// Updates underlying sequence paths to not be under /dry-run directory.
+    pub fn update_paths_to_broadcasted(
+        &mut self,
+        config: &Config,
+        sig: &str,
+        target: &ArtifactId,
+    ) -> Result<()> {
+        match self {
+            ScriptSequenceKind::Single(sequence) => {
+                sequence.paths =
+                    Some(ScriptSequence::get_paths(config, sig, target, sequence.chain, false)?);
+            }
+            ScriptSequenceKind::Multi(sequence) => {
+                (sequence.path, sequence.sensitive_path) =
+                    MultiChainSequence::get_paths(config, sig, target, false)?;
+            }
+        };
+
+        Ok(())
+    }
+}
+
+impl Drop for ScriptSequenceKind {
+    fn drop(&mut self) {
+        self.save(false, true).expect("could not save deployment sequence");
```

Comment:
Should probably be `if let Err(e) = { error!() }` rather than panicking on IO errors